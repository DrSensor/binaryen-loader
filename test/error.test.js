import loader from '../src';

describe('Errors', () => {
  test('Loader Error', () => {
    const err = () => loader.call();

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('Validation Error', () => {
    const err = () =>
      loader.call({
        query: { useRelativePath: 1 },
        emitFile: true,
      });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });
});
