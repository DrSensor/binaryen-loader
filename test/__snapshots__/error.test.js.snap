// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Loader Error 1`] = `"Cannot read property 'version' of undefined"`;

exports[`Errors Validation Error 1`] = `
"binaryen-loader

  [4mOptions Validation Error[24m

  [2moptions[22m['useRelativePath']  [34mis an invalid additional property[39m
"
`;
